lib/network/modules/components/shared/ComponentUtil.js

```diff
@@ -49,6 +65,56 @@ class ComponentUtil {
 
     return value;
   }
+
+
+  /**
+   * Check if the point falls within the given rectangle.
+   *
+   * @param {rect} rect
+   * @param {point} point
+   * @param {rotationPoint} [rotationPoint] if specified, the rotation that applies to the rectangle.
+   * @returns {boolean}  true if point within rectangle, false otherwise
+   * @static
+   */
+  static pointInRect(rect, point, rotationPoint) {
+    if (rect.right <= rect.left || rect.bottom <= rect.top) {
+      return false;  // early out
+    }
+
+    if (rotationPoint !== undefined) {
```

Comment:
I prefer if we invert this if statement and do an early return, and then we can unindent the else case

Reply:
This would mean that the return statement would be doubled in the code. I prefer this construct.